### PR TITLE
Feature/bitbucket porting

### PR DESCRIPTION
**Pull request by terabyte**

### CODE REVIEW

File: src/main/java/com/palantir/stash/stashbot/managers/PluginUserManager.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public PluginUserManager(UserAdminService uas, PermissionAdminService pas, UserS
     }
 
     public void createStashbotUser(JenkinsServerConfiguration jsc) {
-        StashUser user = us.getUserByName(jsc.getStashUsername());
+        ApplicationUser user = us.getUserByName(jsc.getStashUsername());
```
Comment (memory) [inline src/main/java/com/palantir/stash/stashbot/managers/PluginUserManager.java:59]: as long as you're s/StashUser/ApplicationUser/ everywhere, may as well do it to your methods?

File: src/main/java/com/palantir/stash/stashbot/urlbuilder/StashbotUrlBuilder.java [hunk 2]
```diff
@@ -103,4 +108,16 @@ private String mask(String str) {
             return str;
         }
     }
+    
+    public String getJenkinsJobUrl(Repository repo, JobTemplate jt) throws SQLException {
+        RepositoryConfiguration rc = cps.getRepositoryConfigurationForRepository(repo);
+        JenkinsServerConfiguration jsc = cps.getJenkinsServerConfiguration(rc.getJenkinsServerName());
+        // XXX this is broken with folders, I think
```
Comment (memory) [inline src/main/java/com/palantir/stash/stashbot/urlbuilder/StashbotUrlBuilder.java:115]: concerning :)

File: src/main/java/com/palantir/stash/stashbot/managers/PluginUserManager.java (hunk 1 above)
Comment (terabyte) [inline src/main/java/com/palantir/stash/stashbot/managers/PluginUserManager.java:59]: Yeah, maybe I'll fix that in a future change.  Good point, would be less confusing.

File: src/main/java/com/palantir/stash/stashbot/urlbuilder/StashbotUrlBuilder.java (hunk 2 above)
Comment (terabyte) [inline src/main/java/com/palantir/stash/stashbot/urlbuilder/StashbotUrlBuilder.java:115]: Fixed in a future CR.

Comment (terabyte) [inline src/main/java/com/palantir/stash/stashbot/urlbuilder/StashbotUrlBuilder.java:115]: Oh yeah, you already reviewed the fix here: https://github.com/terabyte/stashbot/pull/15

**Pull request merged**